sale_layout_category_hide_detail/views/account_portal_templates.xml

```diff
@@ -0,0 +1,59 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2018 Tecnativa - Ernesto Tejeda
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+<odoo>
+    <template id="portal_invoice_report_layouted_hide_detail" inherit_id="account.portal_invoice_report">
+        <xpath expr="//span[@t-field='l.name']/../../.." position="attributes">
+            <attribute name="class" add="hidden" separator=" "/>
+        </xpath>
+        <xpath expr="//span[@t-field='l.name']/../../.." position="before">
+            <tbody class="o_portal_report_tbody">
+                <t t-foreach="invoice.order_lines_layouted()" t-as="page" name="lines_layouted">
+                    <t t-foreach="page" t-as="layout_category">
+                        <t t-set="lines_subtotal" t-value="layout_category['lines_subtotal']"/>
+                        <t t-if="layout_category_size > 1 or page_size > 1" groups="sale.group_sale_layout">
+                            <tr class="active">
+                                <td t-att-colspan="3 if layout_category['hide_details'] else 7" style="font-weight: bold; border-bottom: 1px solid black;">&amp;bull;
+                                    <t t-esc="layout_category['name']"/>
+                                </td>
+                                <td t-if="layout_category['hide_details']" colspan="4" class="text-right" style="border-bottom: 1px solid black;">
+                                    <strong>Subtotal: </strong>
+                                    <span t-esc="lines_subtotal" t-options="{'widget': 'monetary', 'display_currency': invoice.currency_id}"/>
+                                </td>
+                            </tr>
+                        </t>
+                        <tr t-foreach="layout_category['lines']" t-as="l">
```

Comment:
This is still replacing previous layout. It doesn't matter you hide the other element instead of doing `position="replace"`. The result is the same. I ask about the same problem: what happens if other module adds an extra column to each order/invoice line and show it in the portal? with this, you will lose that column. Please do the same as with reports, changing only the minimum pieces of layout.

This is applicable to whole layout, although I put the comment here in the middle.

Reply:
done, thank you